c++/cmake/CapnProtoTargets.cmake

```diff
@@ -86,7 +86,6 @@ function(_capnp_import_pkg_config_target target)
   find_library(CapnProto_${target}_IMPORTED_LOCATION
     NAMES ${target_name_shared} ${target_name_static}  # prefer libfoo-version.so over libfoo.a
     PATHS ${${target}_LIBRARY_DIRS}
-    NO_DEFAULT_PATH
```

Comment:
@intelfx, I thought about this a bit. As written, if the user has capnproto installed in a custom location and relies on the `PKG_CONFIG_PATH` environment variable to find it, pkg-config and CMake's `find_library` will find different installations, because system-default paths are searched before the `PATHS` argument (and the `HINTS` argument, which is the other one we could use). It is difficult to prohibit or even diagnose this situation, so I think we should change this to:

```cmake
find_library(...
  NAMES ...
  PATHS ...
  NO_DEFAULT_PATH
)
find_library(...
  NAMES ...
)
```

That makes the search order explicit, and is a fairly well-recognized CMake idiom.

I'd be happy to update the PR, if that's okay with you?